core/arch/arm/kernel/thread_private.h

```diff
@@ -180,7 +180,17 @@ void thread_set_irq_sp(vaddr_t sp);
 void thread_set_fiq_sp(vaddr_t sp);
 
 /* Read usr_sp banked CPU register */
-uint32_t thread_a32_get_usr_sp(void);
+uint32_t thread_get_usr_sp(void);
+
+/* Thread safe read usr_sp banked CPU register */
+static inline uint32_t thread_get_usr_sp_safe(void)
+{
+	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
```

Comment:
Is `THREAD_EXCP_FOREIGN_INTR` more appropriate here?

Reply:
Why not implement it all in assembly in `thread_get_usr_sp()`?
Adding a `cpsid aif` after the `mrs` instruction should be enough.

Reply:
good point. thanks.

Reply:
done